airflow/www/static/js/dags.js

```diff
@@ -126,16 +126,75 @@ $(".typeahead").typeahead({
       success: callback,
     });
   },
+  matcher(item) {
+    const it = this.displayText(item);
+    const query = this.query.toLowerCase();
+    const queryValue = query.replace(/^([a-z]+:\s*)/i, "");
+
+    return it.toLowerCase().indexOf(queryValue) !== -1;
+  },
+  displayText(item) {
+    if (typeof item !== "undefined" && typeof item.name !== "undefined") {
+      if (item.type === "task") {
+        return `${item.type}: ${item.name} <i>in ${item.dag_id}</i>`;
+      }
+      return `${item.type}: ${item.name}`;
+    }
+    return item;
+  },
+  sorter(items) {
+    const beginswith = [];
+    const caseSensitive = [];
+    const caseInsensitive = [];
+    let item;
+    let { query } = this;
+    query = query.replace(/^([a-z]+:\s*)/i, "");
```

Comment:
Let's create a new variable for query instead of using `let` and reassigning its value.